cabrerahector/wordpress-popular-posts
# Add: exclude traffic with cookies

## What?

Exclude all internal traffic (log views) with cookies.


## Why?

Every visit made an employee can skews log views data, and becomes an even larger problem when working with larger business.


## How?

Store cookies name to `wpp-json`, check if cookies exists, `do_request = false;`

Fix this issue: [Exclude internal traffic with cookie](https://wordpress.org/support/topic/exclude-internal-traffic-with-cookie/)


## Testing Instructions
1. Open [this site (WP Test by TasteWP)](https://symptomaticbone.s4-tastewp.com/wp-admin/) and login (**automatically deleted after 48 hours**)
  - Username: admin 
  - Password: MesT-4qTcF8
2. Go to **Settings** > **WordPress Popular Posts** > **Tools**
3. Scroll to **Data** > **Exclude log views with cookies** > **Yes**
4. **Cookies field name** > type `internal_user`
5. Open [site post](https://symptomaticbone.s4-tastewp.com/2020/08/05/hello-world/)
6. Open DevTools > Console,
7. Add cookies `internal_user` with this script
```` javascript
function setCookie(cname, cvalue, exdays, cdomain) {
  var d = new Date();
  d.setTime(d.getTime() + (exdays*24*60*60*1000));
  var expires = "expires="+ d.toUTCString();
  document.cookie = cname + "=" + cvalue + ";" + expires + ";path=/ ; domain=" + cdomain;
}
setCookie("internal_user", "internal", 365, location.hostname.replace(/^www\./i, "")); //One year
````
8. Reload the page.
9. Check [wpp stats](https://symptomaticbone.s4-tastewp.com/wp-admin/options-general.php?page=wordpress-popular-posts&tab=stats)
ㅤ
